staramr/results/QualityModule.py

```diff
@@ -0,0 +1,280 @@
+        :param genome_size_lower_bound: The lower bound for the genome size as defined by the user for quality metrics
+        :param genome_size_upper_bound: The upper bound for the genome size as defined by the user for quality metrics
+        :param minimum_N50_value: The minimum N50 value as defined by the user for quality metrics
+        :param minimum_contig_length: The minimum contig length as defined by the user for quality metrics
+        :param unacceptable_num_contigs: The number of contigs in a file, equal to or above our minimum contig length, for which to raise a flag as defined by the user for quality metrics
+        """
+        self._files = files
+        self._genome_size_lower_bound = genome_size_lower_bound
+        self._genome_size_upper_bound = genome_size_upper_bound
+        self._minimum_N50_value = minimum_N50_value
+        self._minimum_contig_length = minimum_contig_length
+        self._unacceptable_num_contigs = unacceptable_num_contigs
+        
+    
+    def _create_quality_module_dataframe(self):
+        """
+        Goes through the files and creates a dataframe consisting of the file's genome length, N50 value and the number of contigs greater or equal to the minimum contig length as
+        specified by the quality metrics. It also consists of the feedback for whether or not the file passed the quality metrics and if it didn't feedback on why it failed
+        :return: A pd.dataframe containing the genome size, N50 value, number of contigs equal to or above our user defined minimum contig length
+        as well as the results of our quality metrics (pass or fail) and the corresponding feedback
+        """
+        name_set=[]
+        for myFile in self._files:
+            name_set.append(path.splitext(path.basename(myFile))[0])
+
+        files_contigs_and_genomes_lengths=self._get_files_contigs_and_genomes_lengths(self._files)
+
+        files_genome_length_feedback = self._get_genome_length_feedback(files_contigs_and_genomes_lengths[1],self._genome_size_lower_bound,self._genome_size_upper_bound)
+        
+        files_N50_value_feedback=self._get_N50_feedback(files_contigs_and_genomes_lengths[0],files_contigs_and_genomes_lengths[1],self._minimum_N50_value)
+        
+        file_num_contigs_over_minimum_bp_feedback= self._get_num_contigs_over_minimum_bp_feedback(files_contigs_and_genomes_lengths[0],self._minimum_contig_length,self._unacceptable_num_contigs)
+        
+        quality_module = self._get_quality_module(files_genome_length_feedback,files_N50_value_feedback[1],file_num_contigs_over_minimum_bp_feedback[1])      
+        quality_module_feedback = quality_module[0]
+        quality_module_result = quality_module[1]
+        
+        quality_module_frame=pd.DataFrame([[t,u,v,w,x,y] for t,u,v,w,x,y in zip(name_set,files_contigs_and_genomes_lengths[1],files_N50_value_feedback[0],file_num_contigs_over_minimum_bp_feedback[0],quality_module_result,quality_module_feedback)],
```

Comment:
This one liner is great to get things done but makes it harder to modify in the future.  Most code is written few times but read many times. Readability will be important in the future.

Can you break it down into a more readable fashion.